hedera-mirror-rest/__tests__/utils.test.js

```
@@ -287,3 +287,268 @@ describe('Utils randomString tests', () => {
+    },
+    {
+      name: extraParamTestName,
+      parsedQueryParams: {
+        "account.id": "0.0.3",
+        "timestamp": "2000"
+      },
+      expectedClause: "account.id IN (?)",
+      expectedValues: ["3"]
+    },
+    {
+      name: multipleEqualsTestName,
+      parsedQueryParams: {"account.id": ["0.0.3", "4"]},
+      expectedClause: "account.id IN (?, ?)",
+      expectedValues: ["3", "4"]
+    },
+    {
+      name: duplicateParamsTestName,
+      parsedQueryParams: {"account.id": ["0.0.5", "5", "eq:0.0.5", "lte:0.0.3", "lte:0.0.3", "gte:0.0.3", "gte:0.0.4"]},
+      expectedClause: "account.id <= ? and account.id >= ? and account.id >= ? and account.id IN (?)",
+      expectedValues: ["3", "3", "4", "5"]
+    },
+
+  ];
+  parseQueryParamTest('Utils parseAccountIdQueryParam - ', testSpecs, (spec) => utils.parseAccountIdQueryParam(spec.parsedQueryParams, "account.id"));
+});
+
+describe('Utils parseTimestampQueryParam tests', () => {
+  const testSpecs = [
+    {
+      name: singleParamTestName,
+      parsedQueryParams: {"timestamp": "1000"},
+      expectedClause: "timestamp = ?",
+      expectedValues: ["1000000000000"]
+    },
+    {
+      name: noParamTestName,
+      parsedQueryParams: {},
+      expectedClause: "",
+      expectedValues: []
+    },
+    {
+      name: multipleParamsTestName,
+      parsedQueryParams: {"timestamp": ["gte:1000", "lt:2000.222", "3000.333333333"]},
+      expectedClause: "timestamp >= ? and timestamp < ? and timestamp = ?",
+      expectedValues: ["1000000000000", "2000222000000", "3000333333333"]
+    },
+    {
+      name: extraParamTestName,
+      parsedQueryParams: {
+        "timestamp": "1000",
+        "fake.id": "2000"
+      },
+      expectedClause: "timestamp = ?",
+      expectedValues: ["1000000000000"]
+    },
+    {
+      name: multipleEqualsTestName,
+      parsedQueryParams: {"timestamp": ["1000", "4000"]},
+      expectedClause: "timestamp = ? and timestamp = ?",
```

This would return 0 rows. We should throw an error or convert to an IN. If neither is trivial we can create a ticket to capture all these other params with multiple values.

---

Will create a ticket for this.  It should be easy to return IN for balance and public key, and credit/debit doesn't have this issue because everything converts to > 0 or < 0, the one I worry about is timestamp because we allow for opOverrides (and in one case we actually do override =), it may require a bigger effort to address that.

---

Opened ticket #1653 to address supporting multiple parameters for all others